src/Adapters/Storage/Mongo/MongoTransform.js

```diff
@@ -487,10 +487,14 @@ function transformTopLevelAtom(atom) {
       return BytesCoder.JSONToDatabase(atom);
     }
     if (GeoPointCoder.isValidJSON(atom)) {
-      return GeoPointCoder.JSONToDatabase(atom);
+      // normally we transform a 'GeoPoint' to just the name when storing to the db, but
+      // _GlobalConfig has no schema so we must preserve type information and other fields
+      return className === "_GlobalConfig" ? atom : GeoPointCoder.JSONToDatabase(atom);
```

Comment:
I'm really not a big fan of using that className hard coded like that in this method that should know nothing about the className.